invenio_users_resources/services/users/service.py

```diff
@@ -74,7 +76,7 @@ def read(self, identity, id_):
         # resolve and require permission
         user = UserAggregate.get_record(id_)
         if user is None:
-            raise LookupError(f"No user with id '{id_}'.")
+            raise UserNotFoundError()
```

Comment:
I think you can keep the message that raised here and consume it in the error handler